src/main/java/Duke.java

```diff
@@ -1,10 +1,112 @@
+import java.util.Scanner;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+
+	public static void main(String[] args) {
+		showWelcomeMessage();
+
+		Task[] taskList = new Task[100];
+		int taskIndex = 0;
+
+		Scanner input = new Scanner(System.in);
+
+		while (true) {
+			String userInput = input.nextLine();
+
+			if ("bye".equalsIgnoreCase(userInput)) {
+				break;
+			}
+
+			String[] commands = userInput.split(" ");
+
+			switch (commands[0]) {
+			case "list":
+				printTaskList(taskList);
+				break;
+
+			case "mark":
+				markSelectedTask(taskList, commands);
+				break;
+
+			case "unmark": {
+				unmarkSelectedTask(taskList, commands);
+				break;
+			}
```

Comment:
I like how you extracted and named methods for the `switch case` - very readable and clean